src/Sylius/Bundle/AdminBundle/Resources/config/services/twig/component.xml

```diff
@@ -260,6 +260,16 @@
             />
         </service>
 
+        <service id="Sylius\Bundle\AdminBundle\Twig\Component\CatalogPromotion\FormComponent">
+            <argument type="service" id="form.factory" />
+            <argument>Sylius\Bundle\AdminBundle\Form\Type\CatalogPromotionType</argument>
+        </service>
+
+        <service id="Sylius\Bundle\AdminBundle\Twig\Component\Product\GenerateProductVariantsFormComponent">
+            <argument type="service" id="form.factory" />
+            <argument>Sylius\Bundle\AdminBundle\Form\Type\ProductGenerateVariantsType</argument>
+        </service>
```

Comment:
These two definitions are duplicated and they are exist above if I see correctly
```suggestion
```